alacritty_terminal/src/tty/windows/child.rs

```diff
@@ -32,7 +33,11 @@ extern "system" fn child_exit_callback(ctx: *mut c_void, timed_out: BOOLEAN) {
     }
 
     let event_tx: Box<_> = unsafe { Box::from_raw(ctx as *mut ChildExitSender) };
-    let _ = event_tx.sender.send(ChildEvent::Exited);
+    let child_handle = event_tx.child_handle.load(Ordering::Relaxed) as HANDLE;
+    let mut exit_code = 0_u32;
+    let status = unsafe { GetExitCodeProcess(child_handle, &mut exit_code) };
+    let exit_code = if status > 0 { Some(exit_code as i32) } else { None };
+    let _ = event_tx.sender.send(ChildEvent::Exited(exit_code));
```

Comment:
Some logical separation of the separate steps with blank lines would be nice here. Otherwise that's just a very dense block that does "everything".